Makd.mak

```diff
@@ -640,19 +640,19 @@ fasttest: $(fasttest)
 
 
 .PHONY: d2conv
-d2conv: $O/d2conv.stamp \
-	$(foreach s,$(SUBMODULES),$O/d2conv/$s.stamp)
+d2conv: $O/d2conv.stamp
 
-$O/d2conv.stamp: $C
-	$(call exec,git ls-files | grep '\.d$$' | \
-		xargs --no-run-if-empty d1to2fix,.,d1to2fix)
-
-$O/d2conv/%.stamp: $C/%
-	$(call exec,cd $<; git ls-files | grep '\.d$$' | \
-		xargs --no-run-if-empty d1to2fix,$<,d1to2fix)
-	$Vmkdir -p $(dir $@)
+$O/d2conv.stamp: $O/d2conv.list
+	$(call exec, d1to2fix --fatal --input=$^, $^)
 	$Vtouch $@
 
+$O/d2conv.list: $(foreach s,$(SUBMODULES), $O/d2conv/$s.list)
+	$Vgit ls-files | { grep '\.d$$' || true; } > $@
+	$Vcat $^ >> $@
```

Comment:
I would use `find_files` here for consistency. Also you could combine the stamp with the list, as the stamp is just an artefact created because the building doesn't produce any real output, but in this case the list file can be used as output.

On the other hand, I think it would make more sense for d1to2fix to support `--stdin` to just get the files via stdin, there is no much of a point on using a file really :D